src/utils/SpokeUtils.ts

```diff
@@ -72,14 +53,36 @@ export async function getBlockRangeForDepositId(
     }
   });
 
+  // Define a mapping of block numbers to number of deposits at that block. This saves repeated lookups.
+  const queriedIds: Record<number, number> = {};
+
+  // Define a llambda function to get the deposit ID at a block number. This function will first check the
+  // queriedIds cache to see if the deposit ID at the block number has already been queried. If not, it will
+  // make an eth_call request to get the deposit ID at the block number. It will then cache the deposit ID
+  // in the queriedIds cache.
+  const _getDepositIdAtBlock = async (blockNumber: number): Promise<number> => {
+    if (queriedIds[blockNumber] === undefined) {
+      queriedIds[blockNumber] = await spokePool._getDepositIdAtBlock(blockNumber);
+    }
```

Comment:
Possible concise alternative ?

```suggestion
    queriedIds[blockNumber] ??= await spokePool._getDepositIdAtBlock(blockNumber);
```